oci/oci.go

```diff
@@ -1 +1,91 @@
 package oci
+
+import (
+	"github.com/pkg/errors"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+// annotation spex from https://github.com/opencontainers/image-spec/blob/master/annotations.md#pre-defined-annotation-keys
+const (
+	separator = `(?:[-._:@+]|--)`
+	alphanum  = `(?:[A-Za-z0-9]+)`
+	component = `(?:` + alphanum + `(?:` + separator + alphanum + `)*)`
+)
+
+var refRegexp = regexp.MustCompile(`^` + component + `(?:/` + component + `)*$`)
+var windowsRefRegexp = regexp.MustCompile(`^([a-zA-Z]:\\.+?):(.*)$`)
+
+// ValidImageName returns true is the specified image matches the open-containers image name specs, false otherwise.
+func ValidImageName(image string) bool {
+	return refRegexp.MatchString(image)
+}
+
+// SplitDirAndImage tries to split the provided OCI reference into the OCI directory and image.
+// The image is returned as string pointer, allowing for nil being returned, if no image part could be extraced from the reference.
+// Neither directory nor image parts are validated at this stage.
+func SplitDirAndImage(reference string) (string, *string) {
+	if runtime.GOOS == "windows" {
+		return splitDirAndImageWindows(reference)
+	}
+	return splitDirAndImageNonWindows(reference)
+}
+
+func splitDirAndImageWindows(reference string) (string, *string) {
+	groups := windowsRefRegexp.FindStringSubmatch(reference)
+	// nil group means no match
+	if groups == nil {
+		return reference, nil
+	}
+
+	// we expect three elements. First one full match, second the capture group for the dir and
+	// the third the capture group for the image
+	if len(groups) != 3 {
+		return reference, nil
+	}
+	return groups[1], &groups[2]
```

Comment:
In non-Windows it is possible to use relative paths, e.g. `oci:somedir:imagename`. This seems to return a non-`nil` image only if the input is an absolute path; is that intentional?

AFAICT relative paths could be allowed without ambiguity:
- `c:\dir\…` cannot mean “the relative path `.\c:\dir`” because `c:` is an invalid directory name
- `c:\dir…` cannot mean “the image `\dir…` in the relative path `c`” because `\dir` is an invalid image name.
- The troublesome case is `c:dir`, which could mean “the relative path `dir` on the current directory of `c:`” or “the image `dir` in the relative path `c`”, but at least https://blogs.msdn.microsoft.com/oldnewthing/20101011-00/?p=12563 says this concept does not exist in Windows anymore.

The above could be non-blocking, left to someone in the future to improve, because the Windows support is new; more troublesome is that this parses `somedir:busybox:latest` as path `somedir:busybox:latest`, missing image, which would then fail validation in `NewReference`; and, worse, `somedir:busybox` as path `somedir:busybox`, no image, which would succeed but be inconsistent with the POSIX version.

Reply:
This is still outstanding I think. Or are you fine with `oci:somedir:busybox` just failing?

Reply:
> and, worse, `somedir:busybox` as path `somedir:busybox`, no image, which would succeed but be inconsistent with the POSIX version.

Sorry, that would pass validation but `somedir:busybox` is an invalid path on Windows, so it would fail either way.

So, overall, with this PR relative paths with images just fail; that’s clearly not worse than before this PR, so non-blocking for me.